api/v1alpha1/akodeploymentconfig_types.go

```diff
@@ -97,26 +97,94 @@ type ExtraConfigs struct {
 	// +optional
 	Log AKOLogConfig `json:"log,omitempty"`
 
-	// Rbac specifies the configuration for AKO Rbac
+	// FullSyncFrequency controls how often AKO polls the Avi controller to update itself
+	// with cloud configurations. Default value is 1800
 	// +optional
-	Rbac AKORbacConfig `json:"rbac,omitempty"`
+	FullSyncFrequency string `json:"fullSyncFrequency,omitempty"`
 
-	// IngressConfigs specifies ingress configuration for ako
+	// ApiServerPort specifies Internal port for AKO's API server for the liveness probe of the AKO pod
+	// default port is 8080
 	// +optional
-	IngressConfigs AKOIngressConfig `json:"ingress,omitempty"`
+	ApiServerPort *int `json:"apiServerPort,omitempty"`
 
 	// DisableStaticRouteSync describes ako should sync static routing or not.
 	// If the POD networks are reachable from the Avi SE, this should be to true.
 	// Otherwise, it should be false.
 	// It would be true by default.
 	// +optional
-	DisableStaticRouteSync bool `json:"disableStaticRouteSync,omitempty"`
+	DisableStaticRouteSync *bool `json:"disableStaticRouteSync,omitempty"`
 
 	// CniPlugin describes which cni plugin cluster is using.
-	// default value should be antrea, set this string if cluster cni is other type.
-	// enum: calico|canal|flannel|openshift|antrea
+	// default value is antrea, set this string if cluster cni is other type.
+	// AKO supported CNI: antrea|calico|canal|flannel|openshift|ncp
+	// +kubebuilder:validation:Enum=antrea;calico;canal;flannel;openshift;ncp
 	// +optional
 	CniPlugin string `json:"cniPlugin,omitempty"`
+
+	// EnableEVH specifies if you want to enable the Enhanced Virtual Hosting Model
+	// in Avi Controller for the Virtual Services, default value is false
+	// +optional
+	EnableEVH *bool `json:"enableEVH,omitempty"`
+
+	// Layer7Only specifies if you want AKO only to do layer 7 load balancing.
+	// default value is false
+	// +optional
+	Layer7Only *bool `json:"layer7Only,omitempty"`
+
+	// NameSpaceSelector contains label key and value used for namespace migration.
+	// Same label has to be present on namespace/s which needs migration/sync to AKO
+	// +optional
+	NameSpaceSelector NameSpaceSelector `json:"namespaceSelector,omitempty"`
+
+	// ServicesAPI specifies if enables AKO in services API mode: https://kubernetes-sigs.github.io/service-apis/.
+	// Currently, implemented only for L4. This flag uses the upstream GA APIs which are not backward compatible
+	// with the advancedL4 APIs which uses a fork and a version of v1alpha1pre1
+	// default value is false
+	// +optional
+	ServicesAPI *bool `json:"servicesAPI,omitempty"`
+
+	// NetworksConfig specifies the network configurations for virtual services.
+	// +optional
+	NetworksConfig NetworksConfig `json:"networksConfig,omitempty"`
+
+	// IngressConfigs specifies ingress configuration for ako
+	// +optional
+	IngressConfigs AKOIngressConfig `json:"ingress,omitempty"`
```

Comment:
This doesn't match the sample we have at `config/samples/network_v1alpha1_akodeploymentconfig.yaml`. Do you think it's a good time to update it?

Reply:
Sure, we should update that yaml file.